crates/router/src/core/payment_methods/cards.rs

```diff
@@ -216,6 +225,208 @@ pub async fn get_or_insert_payment_method(
     }
 }
 
+#[instrument(skip_all)]
+pub async fn get_client_secret_or_add_payment_method(
+    state: routes::AppState,
+    req: api::PaymentMethodCreate,
+    merchant_account: &domain::MerchantAccount,
+    key_store: &domain::MerchantKeyStore,
+) -> errors::RouterResponse<api::PaymentMethodResponse> {
+    let db = &*state.store;
+    let merchant_id = &merchant_account.merchant_id;
+    let customer_id = req.customer_id.clone().get_required_value("customer_id")?;
+
+    if req.card.is_some() || req.bank_transfer.is_some() || req.wallet.is_some() {
+        add_payment_method(state, req, merchant_account, key_store).await
+    } else {
+        let payment_method_id = generate_id(consts::ID_LENGTH, "pm");
+
+        let res = create_payment_method(
+            db,
+            &req,
+            customer_id.as_str(),
+            payment_method_id.as_str(),
+            None,
+            merchant_id.as_str(),
+            None,
+            None,
+            None,
+            key_store,
+            None,
+            Some(enums::PaymentMethodStatus::AwaitingData),
+        )
+        .await?;
+
+        Ok(services::api::ApplicationResponse::Json(
+            api::PaymentMethodResponse {
+                merchant_id: merchant_id.to_string(),
+                customer_id: Some(customer_id),
+                payment_method_id,
+                payment_method: req.payment_method,
+                payment_method_type: req.payment_method_type,
+                card: None,
+                recurring_enabled: false,
+                installment_payment_enabled: false,
+                payment_experience: None,
+                metadata: req.metadata,
+                created: None,
+                bank_transfer: None,
+                last_used_at: None,
+                client_secret: res.client_secret,
+            },
```

Comment:
we should not be referring to fields from the request in response, implement a From / TryFrom from the Response that is generated in the above step